plenum/server/consensus/ordering_service.py

```diff
@@ -1,9 +1,173 @@
 from plenum.common.event_bus import InternalBus, ExternalBus
+from plenum.common.messages.internal_messages import ViewChangeInProgress, LedgerSyncStatus, ParticipatingStatus
+from plenum.common.messages.node_messages import PrePrepare, Prepare, Commit
+from plenum.common.stashing_router import StashingRouter
+from plenum.common.types import f
+from plenum.common.util import compare_3PC_keys
 from plenum.server.consensus.consensus_data_provider import ConsensusDataProvider
+from plenum.server.replica_validator_enums import INCORRECT_INSTANCE, DISCARD, INCORRECT_PP_SEQ_NO, ALREADY_ORDERED, \
+    STASH_VIEW, FUTURE_VIEW, OLD_VIEW, PROCESS, GREATER_PREP_CERT, STASH_CATCH_UP, CATCHING_UP, OUTSIDE_WATERMARKS, \
+    STASH_WATERMARKS
 
 
 class OrderingService:
     def __init__(self, data: ConsensusDataProvider, bus: InternalBus, network: ExternalBus):
         self._data = data
         self._bus = bus
         self._network = network
+        self._stasher = StashingRouter()
+
+        self._stasher.subscribe(PrePrepare, self.process_preprepare)
+        self._stasher.subscribe(Prepare, self.process_prepare)
+        self._stasher.subscribe(Commit, self.process_commit)
+        self._stasher.subscribe_to(network)
+
+        # ToDo: need to define how to get current value (when object instantiating) for next statuses vars
+        self._vc_in_progress = False
+        self._is_participating = False
+        self._is_synced = False
+
+        self._bus.subscribe(ViewChangeInProgress, self.set_vc_in_progress)
+        self._bus.subscribe(LedgerSyncStatus, self.set_is_synced)
+        self._bus.subscribe(ParticipatingStatus, self.set_is_participating)
+
+    def process_preprepare(self, pre_prepare: PrePrepare, sender: str):
```

Comment:
Do we really need `_process_XXX` methods? Maybe it would be more compact and clear to do the logic right in `process_XXX`?
```
        result = self._validator.validate(msg)
        if result != PROCESS:
            return result
        <do process>     
```